Java/flood_fill.java

```diff
@@ -0,0 +1,69 @@
+import java.util.Scanner;
+import java.util.Arrays;
+ 
+/**Flood Fill Class**/
+public class FFill
+{
+   /**Fill Grid Function**/
+    private void fGrid(char[][] a, int x, int y) 
+    {
+        if (a[x][y] == 'X')
+        {
+            a[x][y] = 'W';
+            output(a);
+ 
+            fGrid(a, x + 1, y);
+            fGrid(a, x - 1, y);
+            fGrid(a, x, y + 1);
+            fGrid(a, x, y - 1);
+        }
+    }
+    /**Display Output of the program**/
+    private void output(char[][] a)
+    {
+        System.out.println("\nGrid : ");
+        for (int i = 1; i < a.length - 1; i++)
+        {
+            for (int j = 1; j < a[i].length - 1; j++)
+                System.out.print(a[i][j] +" ");
+            System.out.println();
+        }
+    }
+ 
+   /**Main Class Begain**/ 
+    public static void main(String[] args) 
+    {
+        Scanner scan = new Scanner( System.in );        
+        /**Input Dimensions**/
+        System.out.println("Enter dimensions: ");
+        int M = scan.nextInt();
+        int N = scan.nextInt();
+ 
+        
+        char[][] array = new char[M + 2][N + 2];
+        for (int i = 0; i < M + 2; i++)
+            Arrays.fill(array[i], 'O');
+ 
+        
+        System.out.println("Enter grid with 'X'=passage and 'Y'=obstacle");
+ 
+        for (int i = 1; i < M + 1; i++)
+            for (int j = 1; j < N + 1; j++)
+                array[i][j] = scan.next().charAt(0);    
+ 
+        System.out.println("Enter coordinates: ");
+        int sx = scan.nextInt();
+        int sy = scan.nextInt();
+ 
+        if (arr[sx][sy] != 'X')
+        {
+            System.out.println("Invalid coordinates!");
+            System.exit(0);
+        }
+ 
+        FFill ff = new FFill();
+        ff.fGrid(array, sx, sy);    
+ 
+    }  
+    /**End Main Function**/  
+}
```

Comment:
At the bottom, kindly add Sample I/O, Time and Space Complexity in a multiline comment.
Thank you!

Reply:
Done @sukritishah15 kindly review it.